syntax = "proto3";

package user_service;

message User {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}

message CreateUserRequest {
    string name = 1;
    string email = 2;
    int32 age = 3;
}

message CreateUserResponse {
    User user = 1;
    bool success = 2;
    string message = 3;
}

message GetUserRequest {
    int32 id = 1;
}

message GetUserResponse {
    User user = 1;
    bool success = 2;
    string message = 3;
}

message UpdateUserRequest {
    int32 id = 1;
    string name = 2;
    string email = 3;
    int32 age = 4;
}

message UpdateUserResponse {
    User user = 1;
    bool success = 2;
    string message = 3;
}

message DeleteUserRequest {
    int32 id = 1;
}

message DeleteUserResponse {
    bool success = 1;
    string message = 2;
}

message ListUsersRequest {}

message ListUsersResponse {
    repeated User users = 1;
    bool success = 2;
    string message = 3;
}

service UserService {
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
    rpc GetUser(GetUserRequest) returns (GetUserResponse);
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse);
    rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
}